OpenSim/Simulation/Model/Model.cpp

```diff
@@ -1805,23 +1807,14 @@ void Model::setControls(const SimTK::State& s, const SimTK::Vector& controls) co
 const Vector& Model::getControls(const SimTK::State &s) const
 {
     if( (!_system) || (!_modelControlsIndex.isValid()) ){
-        throw Exception("Model::getControls() requires an initialized Model./n" 
+        throw Exception("Model::getControls() requires an initialized Model./n"
             "Prior call to Model::initSystem() is required.");
     }
 
-    // direct the system shared cache 
-    Measure_<Vector>::Result controlsCache = Measure_<Vector>::Result::getAs(_system->updDefaultSubsystem().getMeasure(_modelControlsIndex));
-
-    if(!controlsCache.isValid(s)){
-        // Always reset controls to their default values before computing controls
-        // since default behavior is for controllors to "addInControls" so there should be valid
-        // values to begin with.
-        controlsCache.updValue(s) = _defaultControls;
-        computeControls(s, controlsCache.updValue(s));
-        controlsCache.markAsValid(s);
-    }
-
-    return controlsCache.getValue(s);
+    // _controlsCache assumes that the controls cache will not be changed or
+    // invalidated between the realization of Stage::Velocity and
+    // Stage::Dynamics
+    return _controlsCache.getValue(s);
```

Comment:
@aseth1 what do you think about this method? Removing the re-evaluation in this method?